libsolidity/codegen/CompilerContext.cpp

```diff
@@ -202,6 +202,8 @@ void CompilerContext::appendInlineAssembly(
 			return false;
 		unsigned stackDepth = _localVariables.end() - it;
 		int stackDiff = _assembly.deposit() - startStackHeight + stackDepth;
+		if (_context == assembly::CodeGenerator::IdentifierContext::LValue)
+			stackDiff -= 1;
```

Comment:
I also wondered for quite a while. It turned out that we have never used inline assembly for code generation (this code is not used for user-supplied inline assembly).